### PR TITLE
Fix configuration update issue 

### LINKED ISSUE #4481: Settings can only be updated one by one
When a user adjust the value of more than one setting on the settings tab and clicks the save button behind one of the updated settings, the values of all other settings reset to their original value. As such, all settings have to be updates one by one at the moment.

[Screencast from 12-23-2022 08:34:09 AM.webm](https://user-images.githubusercontent.com/2684622/209293425-5db54f35-f466-4db4-af78-7abe8118abfe.webm)

### PR DESCRIPTION
**Pull request by matborowczyk**
# Description

* Short description here *

closes #4481 

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [x] Attached issue to pull request
- [x] Changelog entry
- [x] Code is clear and sufficiently documented
- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [x] Correct, in line with design
- [ ] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )

### CODE REVIEW

**Comment by matborowczyk** [comment]
https://user-images.githubusercontent.com/113331659/210999034-5ccd0309-12f3-4254-b7c4-1c8f320a6ca3.mov

**Comment by matborowczyk** [comment]
There is still one small issue, reseting inputs still appear in nested objects, as I couldn't find solution that doesn't include rearranging whole flow, and I want to discuss it on Monday on sprint planning as Laura wanted to discuss UX of that Tab anyway

File: src/Slices/Settings/UI/Tabs/Configuration/Provider.tsx [hunk 1]
```diff
@@ -8,10 +9,42 @@ interface Props {
   settings: EnvironmentSettings.EnvironmentSettings;
 }
 
+function reducer(
```
Comment (LauraStordeur) [inline src/Slices/Settings/UI/Tabs/Configuration/Provider.tsx:12]: Can you add a comment as to Why the reducer was needed ? For later reference, cause I already forgot parts of the explanation from last week x)

Comment (matborowczyk) [reply src/Slices/Settings/UI/Tabs/Configuration/Provider.tsx:11]: Done :)

**Review by LauraStordeur — APPROVED** [review]

**Pull request closed without merging**